components/metal/MetalGrid.vue

```diff
@@ -34,158 +34,143 @@
+  const centralColumn = Math.floor(noDecoherenceColumnCount / 2)
+  const [rowStart, rowEnd] = [1, pattern.length + 1]
+  const [columnStart, columnEnd] = [
+    centralColumn - pattern[0].length / 2,
+    centralColumn + pattern[0].length / 2
+  ]
+
+  function* gen (): Iterable<string> {
+    for (let r = rowStart; r < rowEnd; r++) {
+      for (let c = columnStart; c < columnEnd; c++) {
+        const cellIsAlwaysVisible = pattern[r - rowStart][c - columnStart] === 0
+        if (cellIsAlwaysVisible) {
+          continue
         }
+        yield self.posId({ c, r })
       }
     }
-
-    return gen()
-  })())
-
-  isHidden (pos: CellSpecification) {
-    return this.hiddenCells.includes(this.posId(pos))
   }
 
-  isTrigger (pos: CellSpecification) {
-    const centralColumn = Math.floor(this.noDecoherenceColumnCount / 2)
-    const { x: triggerX, y: triggerY } = this.triggerPositionFromTopCenter
-    const { c, r } = pos
-    return c === centralColumn + triggerX && r === triggerY
-  }
+  return gen()
+})()))
 
-  triggerAnimation (pos: CellSpecification) {
-    if (!this.isTrigger(pos)) { return }
-    this.slotContainerIsHidden = true
-    this.removeCell()
-  }
+const isHidden = computed((pos: CellSpecification) => {
+  return hiddenCells.includes(posId(pos))
+})
 
-  removeCell () {
-    const length = this.fallingCells.length
-    const noMoreCells = length === 0
+const isTrigger = computed((pos: CellSpecification) => {
+  const centralColumn = Math.floor(noDecoherenceColumnCount / 2)
+  const { x: triggerX, y: triggerY } = triggerPositionFromTopCenter
+  const { c, r } = pos
+  return c === centralColumn + triggerX && r === triggerY
+})
 
-    if (noMoreCells) {
-      setTimeout(() => {
-        this.$router.push({ path: '/metal' })
-      }, this.timeToLoadMetal)
-      return
-    }
+const triggerAnimation = (pos: CellSpecification) => {
+  if (!isTrigger.value(pos)) { return }
+  slotContainerIsHidden.value = true
+  removeCell()
+}
 
-    const index = Math.floor(Math.random() * length)
-    const cellToHideId = this.fallingCells.splice(index, 1)[0]
-    this.hiddenCells.splice(0, 0, cellToHideId)
+function removeCell () {
+  const length = fallingCells.value.length
+  const noMoreCells = length === 0
 
-    setTimeout(this.removeCell, this.timeToRemoveNextCell)
+  if (noMoreCells) {
+    setTimeout(() => {
+      this.$router.push({ path: '/metal' })
+    }, timeToLoadMetal)
+    return
   }
 
-  posId (pos: CellSpecification): string {
-    return `cell-${pos.c}-${pos.r}`
-  }
+  const index = Math.floor(Math.random() * length)
+  const cellToHideId = fallingCells.value.splice(index, 1)[0]
+  hiddenCells.splice(0, 0, cellToHideId)
 
-  rowId (index: number): string {
-    return `row-${index}`
-  }
+  setTimeout(removeCell, timeToRemoveNextCell)
+}
+
+const posId = (pos: CellSpecification): string => {
+  return `cell-${pos.c}-${pos.r}`
+}
+
+const rowId = (index: number): string => {
+  return `row-${index}`
```

Comment:
Same as before: function definition. In addition, I would suggest to change the function names to:
- getPosId
- getRowId

Reply:
Agree. Changed in https://github.com/Qiskit/qiskit.org/pull/2935/commits/8528b816e07a7f7e71d0860733ba1431cdb149ba.